w6/assignment6_1_2/character.cpp

```diff
@@ -0,0 +1,118 @@
+#include "character.h"
+#include <iostream>
+#include <cassert>
+
+void Character::set_tag(Tag tag)
+{
+    this->tag = tag;
+}
+
+const Tag& Character::get_tag() const
+{
+    return tag;
+}
+
+const std::string Character::get_identity()
+{
+    if(tag == Tag::robot)
+    {
+        return std::to_string(id);
+    }
+    else if(tag == Tag::person)
+    {
+        return std::string{first_letter};
+    }
+    else if(tag == Tag::alien)
+    {
+        return std::to_string(wavelength);
+    }
+    else
+    {
+        return "Error: identity not found";
+    }
+}
+
+void Character::set(int id)
+{
+    if(tag == Tag::robot)
+    {
+        this->id = id;
+    }
+    else
+    {
+        set_tag(Tag::not_found);
+    }
+}
+
+void Character::set(char first_letter)
+{
+    if(tag == Tag::person)
+    {
+        this->first_letter = first_letter;
+    }
+    else
+    {
+        set_tag(Tag::not_found);
+    }
+}
+
+void Character::set(float wavelength)
+{
+    if(tag == Tag::alien)
+    {
+        this->wavelength = wavelength;
+    }
+    else
+    {
+        set_tag(Tag::not_found);
+    }
+}
+
+const int& Character::get_id() const
```

Comment:
Check inside these that the correct type is trying to be acquired.  (e.g. throw if the tag is not ID when trying to get_id)